core/src/test/java/nl/nn/adapterframework/pipes/PasswordHashPipeTest.java

```diff
@@ -0,0 +1,63 @@
+package nl.nn.adapterframework.pipes;
+
+import nl.nn.adapterframework.core.PipeLineSessionBase;
+import nl.nn.adapterframework.core.PipeRunResult;
+import nl.nn.adapterframework.util.PasswordHash;
+import org.junit.Test;
+import org.junit.Before;
+import org.junit.After;
+import org.mockito.Mock;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * PasswordHashPipe Tester.
+ * @author <Sina>
+ */
+public class PasswordHashPipeTest extends PipeTestBase<PasswordHashPipe> {
+
+    @Mock
+    private PipeLineSessionBase session = new PipeLineSessionBase();
+
+    @Override
+    public PasswordHashPipe createPipe() {
+        return new PasswordHashPipe();
+    }
+
+
+    /**
+     * Method: doPipe(Object input, IPipeLineSession session)
+     */
+    @Test
+    public void testHashPipe() throws Exception {
+        session.put("key", "3:2342:2342" );
+        pipe.configure();
+        PipeRunResult res = doPipe(pipe, "password", session);
+        assertEquals("success", res.getPipeForward().getName());
+    }
+
+    @Test
+    public void testValidatePipe() throws Exception {
+
+        String sc = PasswordHash.createHash("password");
```

Comment:
Maybe you can also convert the current PasswordHash tests (in the static main method of the class) to proper junit tests.

Reply:
Okay, I thought you were talking about the PasswordHashPipe rather than PasswordHash class. Checking it out now.

Reply:
Converted the main method into a test case.